presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotBrokerPageSource.java

```diff
@@ -315,6 +316,12 @@ public int populateFromPqlResults(
         if (exceptions != null && exceptions.isArray() && exceptions.size() > 0) {
             // Pinot is known to return exceptions with benign errorcodes like 200
             // so we treat any exception as an error
+            if (PinotSessionProperties.isMarkDataFetchExceptionsAsRetriable(session)) {
+                throw new PinotException(
+                    PINOT_DATA_FETCH_EXCEPTION,
+                    Optional.of(pql),
+                    String.format("Query %s encountered retriable exception %s", pql, exceptions.get(0)));
+            }
             throw new PinotException(
```

Comment:
I think we can unconditionally throw PINOT_DATA_FETCH_EXCEPTION here, if it is optionally marked as retriable elsewhere. ie, you can just make that change on line 319/326 below

Reply:
I wish I could set it accordingly, but cannot find a way.
Also I prefer making this config as a session config, so user can turn it on/off along with retry count without restart presto.

Reply:
Fair enough. Can we do this on line 319/326 then ? It would be nice to keep as much of the code common if possible :-)

```
throw new PinotException(
      PinotSessionProperties.isMarkDataFetchExceptionsAsRetriable(session) ? PINOT_DATA_FETCH_EXCEPTION : PINOT_EXCEPTION,
                    Optional.of(pql),
                    String.format("Query %s encountered exception %s", pql, exceptions.get(0)));
```

Reply:
done